### ISSUE TITLE
PERF402 slightly incorrect fix suggestion

### ISSUE THREAD

**Issue by Skylion007** [astral-sh/ruff#8070]
```
a = []
for i in range(10):
    for j in range(100):
        a.append(j)
```
ruff version 0.1.0

This rule outputs
`test-perf.py:4:9: PERF402 Use `list` or `list.copy` to create a copy of a list`
Which while correct in that something should be done here, it really should be calling .extend() not the list or list.copy() constructor. Doing
```
a = []
for i in range(10):
    a = list(range(100))
```
would yield an invalid result.

**Comment by qdegraaf** [comment]
Do we want to specify that this is only really relevant for nested if-loops? Or do we also want to suggest it for single loops like in the default example in our docs and [upstream implementation](https://github.com/tonybaloney/perflint#w8402--use-a-list-copy-instead-of-a-for-loop-use-list-copy).

Right now PERF402 docs describe converting from:

```python
/// original = list(range(10000))
/// filtered = []
/// for i in original:
///     filtered.append(i)
```
to
```python
/// original = list(range(10000))
/// filtered = list(original)
```
([NIT] that filtered name is kind of misleading too here)
Do we want this to become:

```python
/// original = list(range(10000))
/// filtered = [].extend(original)
```
?
